src/index.test.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { ReactNode } from "react";
+import { render, fireEvent, act, screen } from "@testing-library/react";
+import { ReactSearch } from ".";
+import * as SearchInput from "./SearchInput";
+import * as SearchResult from "./SearchResult";
+import { useSearch } from "./useSearch";
+
+jest.useFakeTimers();
+
+jest.mock("./SearchModal", () => {
+  return {
+    SearchModal: ({ children }: { children: ReactNode[] }) => <div>{children}</div>
+  };
+});
+jest.mock("./useSearch", () => ({
+  useSearch: jest.fn()
+}));
+
+afterAll(() => jest.useRealTimers());
+
+describe("ReactSearch", () => {
+  it("should render a SearchInput", () => {
+    (useSearch as jest.Mock).mockImplementation(() => ({
+      fetchSearchResults: jest.fn(),
+      isLoading: false
+    }));
+    const searchInputSpy = jest.spyOn(SearchInput, "SearchInput");
+
+    renderSearch();
+
+    expect(searchInputSpy).toHaveBeenCalledTimes(1);
```

Comment:
We're not really testing that ReactSearch renders SearchInput... just that it calls the SearchInput function. What do we really care about, though? It seems to me that we should really care about the behavior of RenderSearch. E.g. interacting with the search input and viewing the search results. Right?

Reply:
As we use functional components, checking that the function is called should be good enough to see that verify that it is rendered.

And then, because we test the SearchInput and SearchResult components separately, at this higher level we need only test that the functional components are called with the right props. 

This is because:
- <Search /> is composed of <SearchInput> and instances of <SearchResult> (which IMO now should probably be a <SearchResultsList> component that accepts a selectedIndex prop
- We verify the behavior of <SearchInput> through its own unit test
- We also verify the behavior of <SearchResult> through its own unit test

I will agree however that this is a special case, since the navigation with up/down arrows is ultimately triggered by the input and is visually represented in search results. I can definitely follow up with a test for that.

IMO this is the tricky thing with unit tests, i.e. how far do you go without venturing into the area of integration tests. How far down the component hierarchy do you render, and at what point do you mock?

In my experience, what I've found effective is to approach it from the perspective of: "What if these child components were imported from an external library like Chakra or Mantine?" In most cases, the unit tests would just verify that the parent component renders them with the right props.